scripts/export.py

```diff
@@ -89,11 +100,39 @@ def nemo_export(argv):
     nemo_in = args.source
     out = args.out
 
+    # Create a PL trainer object which is required for restoring Megatron models
+    cfg_trainer = TrainerConfig(
+        gpus=1,
+        accelerator="ddp",
+        num_nodes=1,
+        # Need to set the following two to False as ExpManager will take care of them differently.
+        logger=False,
+        checkpoint_callback=False,
+    )
+    trainer = pl.Trainer(cfg_trainer)
+
     logging.info("Restoring NeMo model from '{}'".format(nemo_in))
     try:
         with torch.inference_mode():
-            # Restore instance from .nemo file using generic model restore_from
-            model = ModelPT.restore_from(restore_path=nemo_in)
+            # If the megatron based NLP model was trained on NeMo < 1.5, then we need to update the lm_checkpoint on the model config
```

Comment:
This megatron_legacy block would be better off as a separate script that converts legacy .nemo to regular .nemo, and not part of the export script. Otherwise I would have to duplicate this logic in nemo2riva if we want to support legacy megatron -> Riva link.

Reply:
Introduced a new file scripts/legacy_megatronbert_nlp_to_current_version.py to convert legacy MegatronBert based checkpoints to uptodate version of NeMo
Reverted back export.py to it's previous state
Now developers have to run scripts/legacy_megatronbert_nlp_to_current_version.py first and then scripts/export.py

Reply:
@ramanathan831 can we move export scripts to its own directory: `scripts/export/...` ?